### PR TITLE
feat: changed the provider of the Allocator API from web3 to ethers.js

### PR DESCRIPTION
**Pull request by k3nt0w**
Fixes #

## Proposed Changes

- I changed the provider of the Allocator API from web3 to ethers.js
- I mocked up the calculateMaxRewardsPerBlock function in index.spec.ts for rigorous unit testing.

```sh
~/s/g/d/dev-kit-js ❯❯❯ yarn test lib/allocator
yarn run v1.22.10
$ jest lib/allocator
 PASS  lib/allocator/index.spec.ts
 PASS  lib/allocator/calculateMaxRewardsPerBlock.spec.ts

Test Suites: 2 passed, 2 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        1.853 s
Ran all test suites matching /lib\/allocator/i.
✨  Done in 3.35s.
```

### CODE REVIEW

File: lib/allocator/index.spec.ts [hunk 1]
```diff
@@ -1,41 +1,39 @@
-import Web3 from 'web3'
-import { CustomOptions } from '../option'
+import { ethers } from 'ethers'
 import { createAllocatorContract, CreateAllocatorContract } from '.'
 import { createCalculateMaxRewardsPerBlockCaller } from './calculateMaxRewardsPerBlock'
 import { allocatorAbi } from './abi'
 
+jest.mock('./calculateMaxRewardsPerBlock')
```
Comment (k3nt0w) [inline lib/allocator/index.spec.ts:6]: @aggre 
I thought that calling the calculateMaxRewardsPerBlock function in this test would not be appropriate as a unit test, so I mocked it. What do you think?

Comment (k3nt0w) [reply lib/allocator/index.spec.ts:6]: @abyssparanoia 
I would like to hear your opinion as well.

File: lib/allocator/index.spec.ts [hunk 2]
```diff
@@ -1,41 +1,39 @@
...
-			client.setProvider(new Web3.providers.HttpProvider(host))
-
-			const expected: (
-				address?: string,
-				options?: CustomOptions
-			) => CreateAllocatorContract = (
-				address?: string,
-				options?: CustomOptions
+			const address = 'address'
+			const provider = new ethers.providers.JsonRpcProvider(host)
```
Comment (k3nt0w) [inline lib/allocator/index.spec.ts:16]: @aggre 
Although it is test code, we want to make sure that there are no major changes in the behavior of the program as much as possible. So I used `ethers.providers.JsonRpcProvider` , is that OK?

File: lib/allocator/index.spec.ts (hunk 1 above)
Comment (abyssparanoia) [reply lib/allocator/index.spec.ts:6]: I think this is good

**Pull request merged**